File: standard-telegram-bot/src/main/java/com/immmus/standard/telegram/bot/settings/DefaultWebHookBotSettings.java
package com.immmus.standard.telegram.bot.settings;

import java.util.Optional;

final public class DefaultWebHookBotSettings implements WebHookBotSettings {
    private final TelegramBotSettings botSettings;
    private final String webHoopPath;

    public DefaultWebHookBotSettings(String webHoopPath, TelegramBotSettings botSettings) {
        this.botSettings = botSettings;
        this.webHoopPath = webHoopPath;
    }

    public DefaultWebHookBotSettings(String webHoopPath, String botUsername, String token) {
        this(webHoopPath, new TelegramBotSettings.WithoutProxyDefaultTelegramBotSettings(botUsername, token));
    }

    public DefaultWebHookBotSettings(String webHoopPath, String botUsername, String token, ProxySettings proxySettings) {
        this(webHoopPath, new TelegramBotSettings.WithProxyDefaultTelegramBotSettings(botUsername, token, proxySettings));
    }

    @Override
    public String webHookPath() {
        return this.webHoopPath;
    }

    @Override
    public String botUsername() {
        return this.botSettings.botUsername();
    }

    @Override
    public String token() {
        return this.botSettings.token();
    }

    @Override
    public Optional<ProxySettings> proxySettings() {
        return this.botSettings.proxySettings();
    }
}

File: standard-bot/src/main/kotlin/com/immmus/common/telegram/bot/settings/DefaultWebHookBotSettings.kt
package com.immmus.common.telegram.bot.settings

class DefaultWebHookBotSettings(
    override val webHookPath: String,
    private val botSettings: TelegramBotSettings
) : WebHookBotSettings {

    constructor(webHookPath: String, botUsername: String, token: String) :
            this(
                webHookPath,
                TelegramBotSettings.WithoutProxyDefaultTelegramBotSettings(
                    botUsername,
                    token
                )
            )

    constructor(webHookPath: String, botUsername: String, token: String, proxySettings: ProxySettings) :
            this(
                webHookPath,
                TelegramBotSettings.WithProxyDefaultTelegramBotSettings(
                    botUsername,
                    token,
                    proxySettings
                )
            )

    override fun botUsername(): String {
        return botSettings.botUsername()
    }

    override fun token(): String {
        return botSettings.token()
    }

    override fun proxySettings(): ProxySettings? = botSettings.proxySettings()
}